onmt/utils/misc.py

```diff
@@ -72,11 +72,11 @@ def tile(x, count, dim=0):
     perm = list(range(len(x.size())))
     if dim != 0:
         perm[0], perm[dim] = perm[dim], perm[0]
-        x = x.permute(perm).contiguous()
+        x = x.permute(perm)
```

Comment:
Is this change necessary?
If there is a break in terms of memory layout, shouldn't we fix it at the nearest place possible?
Leave a tensor variable to stay broken rather than fix it right away feels worrying.

Reply:
Does it matter that the operation is not done at this point ? I will be performed in the next lines. I can remove it but I will have to perform the second contiguous operation in this function.
The tile function is/should not be called more than once per vector and per line of inference.

Reply:
No, it does not matter actually. But I personally prefer the array to stays in good condition at any point in the function if such change does not have any benefits in terms of performance.
Here is a nice explanation for [contiguous or non-contiguous](https://discuss.pytorch.org/t/contigious-vs-non-contigious-tensor/30107/2).